toolchain/install/busybox_info_test.cpp

```diff
@@ -166,5 +167,58 @@ TEST_F(BusyboxInfoTest, NotBusyboxSymlink) {
   EXPECT_FALSE(info.ok());
 }
 
+TEST_F(BusyboxInfoTest, LayerSymlinksInstallTree) {
```

Comment:
Yesterday, one of the concerns I mentioned was incorrect install selection. You'd said you're okay with that, but would it be worth adding a test that demonstrates the expected result?

(note, you could share install tree creation via a helper with `LayerSymlinksInstallTree`)

e.g., choosing "/opt" because that's a thing:

```
TEST_F(BusyBoxInfoTest, SymlinkAtSameLevelReturnsWrongInstall) {
  // Just enough of the system install that incorrect looks can find the busybox.
  MakeDir(dir_ / "lib");
  MakeDir(dir_ / "lib/carbon");
  MakeFile(dir_ / "lib/carbon/carbon-busybox");

  // What it would look like if someone also did an install in "opt".
  MakeDir(dir_ / "opt");
  MakeDir(dir_ / "opt/lib");
  MakeDir(dir_ / "opt/lib/carbon");
  MakeFile(dir_ / "opt/lib/carbon/carbon-busybox");
  MakeDir(dir_ / "opt/lib/carbon/llvm");
  MakeDir(dir_ / "opt/lib/carbon/llvm/bin");
  MakeSymlink(dir_ / "opt/lib/carbon/llvm/bin/clang", "../../carbon-busybox")
  MakeDir(dir_ / "opt/bin");
  MakeSymlink(dir_ / "opt/bin/carbon", "../lib/carbon/carbon-busybox")

  auto target = MakeSymlink(dir_ / "opt/carbon", "bin/carbon");

  auto info = GetBusyboxInfo(target.string());
  ASSERT_TRUE(info.ok()) << info.error();
  EXPECT_THAT(info->bin_path, Eq(dir_ / "opt/../lib/carbon/carbon-busybox"));
  EXPECT_THAT(info->mode, Eq("clang"));
}
```

Reply:
Good call. I've framed it more as a "positive" test that the busybox info finds a relative busybox before walking symlinks further. I think this reflects what you're looking for, but tell me if not.

Reply:
Per discussion, I was trying to note the case of "/bin/carbon" versus "/opt/carbon" getting mixed up when "/opt/carbon" is pointing at an install under "/opt" instead of "/". I think the test is doing something different.

Reply:
I see, I've added dedicated testing for this scenario and checks to make sure we only do the relative path walk when in a `bin` directory so its at least less likely to bump into this.